scio-avro/src/main/scala/com/spotify/scio/avro/SpecificRecordAvroSCollection.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.scio.avro
+
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.io.Tap
+import com.spotify.scio.values.SCollection
+import org.apache.avro.file.CodecFactory
+import org.apache.avro.specific.SpecificRecordBase
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
+final class SpecificRecordAvroSCollection[T <: SpecificRecordBase](
```

Comment:
I think we can avoid having this class ... we can make a gen type constrain on the method `saveAsAvroFile` in `AvroSCollection`

Reply:
Actually, since we care about backwards compatibility, we cant do it as i thought because we will end up having method overload with default params. I think the current solution is ok and i think we should revise the other methods.